### ISSUE TITLE
Any plans for CodeActions?

### ISSUE THREAD

**Issue by albfan** [georgewfraser/java-language-server#94]
I see no use of CodeActionKind, not sure if there're any plans to implement it. Missing:

- refactor/extract
- refactor/inline

mostly

**Comment by georgewfraser** [comment]
Yes those would be amazing. Is there any pre-existing java refactoring tool out there that we could piggyback off? I spent some time studying [Refaster](https://errorprone.info/docs/refaster) but it's pretty tangled up with Google's error-prone compiler plugin.

**Comment by albfan** [comment]
Looks promising!. Looking at examples, (about detect and replace `equals("")` with `isEmpty()`) add a codeaction "replaceWithIsEmpty" should be possible. Not sure about how detect action is or not available.

Not sure too how bazel uses it, I will try to use with maven without adding anything in the project config (pom.xml) if possible.

Right now anything will work, later we can polish details.
